internal/provider/resource_cluster.go

```diff
@@ -0,0 +1,354 @@
+/* Copyright 2023 VMware, Inc.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/terraform-provider-vcf/internal/cluster"
+	"github.com/vmware/terraform-provider-vcf/internal/datastores"
+	"github.com/vmware/terraform-provider-vcf/internal/network"
+	validation_utils "github.com/vmware/terraform-provider-vcf/internal/validation"
+	"github.com/vmware/vcf-sdk-go/models"
+	"strings"
+)
+
+func ResourceCluster() *schema.Resource {
+	clusterResourceSchema := clusterSubresourceSchema().Schema
+	clusterResourceSchema["domain_id"] = &schema.Schema{
+		Type:         schema.TypeString,
+		Required:     true,
+		Description:  "The ID of a domain that the cluster belongs to",
+		ValidateFunc: validation.NoZeroValues,
+	}
+
+	return &schema.Resource{
+		CreateContext: resourceClusterCreate,
+		ReadContext:   resourceClusterRead,
+		UpdateContext: resourceClusterUpdate,
+		DeleteContext: resourceClusterDelete,
+		Schema:        clusterResourceSchema,
+	}
+}
+
+func resourceClusterDelete(ctx context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	return nil
+}
+
+func resourceClusterUpdate(ctx context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	return nil
+}
+
+func resourceClusterRead(ctx context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	return nil
+}
+
+func resourceClusterCreate(ctx context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	return nil
+}
+
+// clusterSubresourceSchema this helper function extracts the Cluster schema, so that
+// it's made available for merging in the Domain resource schema.
+func clusterSubresourceSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Cluster ID",
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Name of the new cluster that will be added to the specified workload domain",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"host": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "List of vSphere host information from the free pool to consume in the workload domain",
+				MinItems:    1,
+				Elem:        cluster.CommissionedHostSchema(),
```

Comment:
Can we call this "HostSpecSchema" ... to match the API - https://developer.vmware.com/apis/vcf/5.0.0/data-structures/HostSpec/